src/components/Tabs/Tabs.tsx

```diff
@@ -292,23 +288,19 @@ export const Tabs = ({
     scrollableRight && styles['tabs--scrollable-right'],
   );
 
-  const childrenWithProps = React.Children.map(
-    tabs(),
-    // TODO: improve `any` type
-    (child: any, i: number) => {
-      // Checking isValidElement is the safe way and avoids a typescript
-      // error too.
-      if (React.isValidElement(child)) {
-        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-        // @ts-expect-error TODO: fix "No overload matches this call" error
-        return React.cloneElement<Props>(child, {
-          id: idVar[i],
-          ['aria-labelledby']: ariaLabelledByVar[i],
-        });
-      }
-      return child;
-    },
-  );
+  const childrenWithProps = React.Children.map(tabs(), (child, i) => {
```

Comment:
Formatting changed, but the actual change here is removing the explicit types from `child` and `i`. TypeScript now understands what those are without us specifying them.